### ISSUE TITLE
Support remote Docker context

### ISSUE THREAD

**Issue by zcourts** [avast/gradle-docker-compose-plugin#367]
Docker supports adding multiple [contexts](https://docs.docker.com/engine/reference/commandline/context_create/), including [via SSH](https://www.docker.com/blog/how-to-deploy-on-remote-docker-hosts-with-docker-compose/).

Since `localhost` is hard coded at
https://github.com/avast/gradle-docker-compose-plugin/blob/main/src/main/groovy/com/avast/gradle/dockercompose/DockerExecutor.groovy#L148

it's impossible for the plugin to work with a remote context.
Internally, the plugin can check if docker is using default, default-linux or rootless contexts and if it is use localhost as it does now but if not, inspect the context and get the host https://docs.docker.com/engine/reference/commandline/context_inspect/ or some such. 

The error message is also quite misleading, it is not at all clear it comes from the plugin and I spent quite some time debugging thinking it was an issue with docker or compose.

**Comment by psibre** [comment]
Found this issue while running into timeout problems after switching Docker to [rootless mode](https://docs.docker.com/engine/security/rootless/).

Turns out the only relevant hard-coding of [`localhost` in the file linked to above is actually the fallback for when there is no `DOCKER_HOST` environment variable](https://github.com/avast/gradle-docker-compose-plugin/blob/01a21778691bb418439cbb3da7b7cb22ab438ac8/src/main/groovy/com/avast/gradle/dockercompose/DockerExecutor.groovy#L110-L112).

Which is actually the solution, at least to my problem.

# TL;DR

Ensure that `DOCKER_HOST` is exported correctly.
For local Docker in rootless mode, run

    export DOCKER_HOST=unix:///run/user/$(id -u)/docker.sock

Adapted from [the docs](https://docs.docker.com/engine/security/rootless/#install).

**Comment by psibre** [comment]
Perhaps this issue can be closed?
